SEED-platform/seed
# Comstock support

#### Any background context you want to provide?
Please review the code changes to views/column_list_settings.py, there may be a more elegant way to inject the ComStock list setting

#### What's this PR do?
- Adds an opt-in at the organization level to Enable ComStock Functionality, with ComStock column mappings and a protected list settings view
- Switched the UI from using the deprecated PATCH to PUT for /api/v2/columns/X/
- Updates AngularJS to v1.8.0

#### How should this be manually tested?
- `npm install` and `manage.py migrate` 
- Check the `Enable ComStock Functionality` checkbox on your organization settings page
- Go to your organization’s Column Settings page and make sure there’s a new column called ComStock Mapping
- Map some columns to ComStock columns.  Attempt to set two columns to the same ComStock column (duplicates are not allowed).  Save, and then switch to the other property/taxlot tab and map a column to one that you already mapped and then save (it should succeed and remove the mapping from the other table).
- When viewing your inventory check that the list settings profile “ComStock” exists
- Attempt to change/rename/delete the ComStock list setting, there should be a UI notification explaining that it’s protected.  This can also be done by attempting to drag columns on the inventory list view
- Test `GET /api/v2/column_list_settings/` and ensure that the ComStock list setting was injected at the end with the ID null.
- Test `GET /api/v2/column_list_settings/null/` and ensure the ComStock list setting is returned.
- Disable the ComStock functionality and make sure that the opt-in features are no longer available

#### What are the relevant tickets?
#2242
